tutorials/install-github-actions-runner-mac/index.mdx

```diff
@@ -0,0 +1,112 @@
+tags: mac m1 github-actions
+categories: 
+  - bare-metal
+  - apple-silicon
+dates:
+  validation: 2023-08-11
+  posted: 2023-08-11
+---
+
+<Macro id="iam-requirements" />
+
+<Message type="requirement">
+  You have:
+  - You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
+  - You have [created a Mac mini M1](/bare-metal/apple-silicon/how-to/create-mac-mini-m1/)
+  - You have a Github repository with administrator rights.
+</Message>
+
+## Introduction
+
+GitHub Actions is a CI/CD platform that allows users to automate their software development workflows, connected to a Github organization or repository. While GitHub offers online runners with a pay-as-you-go model, self-hosted runners grant increased control over your CI/CD setup and enhanced customization. This is because you manage the runners yourself, rather than relying on GitHub's infrastructure. This tutorial shows you how to set up, configure, and connect a self-hosted runner on a Mac mini M1 to execute macOS pipelines.
+    <Message type="important">
+      Github [recommends](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#self-hosted-runner-security) to only use self-hosted runners with private repositories for security reasons.
+    </Message>
+
+## Installing the runner
+
+1. Navigate to the main page of your [Github](https://github.com/) repository and click **Settings**.
+2. Click **Actions**, located under **Code and automation** on the left sidebar. A menu displays.
+3. Click *Runners*. 
+4. Click the **New self-hosted runner** button. You are redirected to the self-hosted runners configuration page. 
+5. Select **macOS** as a runner image and **ARM64** as an architecture.
+    <Message type="note">
+        After selecting the runner image and architecture, GitHub prompts you to execute a sequence of shell commands. Those commands serve to authenticate your machine with your Github repository, download the necessary runtime and set-up the runner within your environment. Take note of these commands, as you will need to execute them on your Mac mini M1.
+    </Message>
+
+6. [Connect](/bare-metal/apple-silicon/how-to/connect-to-mac-mini-m1/) to your Mac mini M1 using VNC. In this instance, SSH cannot be used as the runner service needs a graphical session to be started. 
+7. Open a terminal and execute the commands previously prompted by Github. Don't launch the runner yet.
+
+    <Message type="note">
+      This tutorial assumes that you picked the default settings when prompted during step 6.
+    </Message>
+
+## Configuring the runner
+
+Once the runner is installed, you must configure it to use the dependencies needed for your workflows.
+
+1. Install the dependencies needed for your workflows. For example, [Homebrew](https://brew.sh/index_fr) is a popular package manager for macOS that can be installed with the following command:
```

Comment:
Maybe we could add having installed a package manager as a prerequesite + add that for the sake of this tutorial, we are using Homebrew.

Reply:
@farhakennar Not sure what you mean here. Could you describe more in depth or make a code suggestion? thanks

Reply:
Sure 😊! I just added Homebrew as a requirement. You can thus delete this first step as we assume the user already has a package manager installed.